pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -304,4 +305,7 @@ def _reorder_callbacks(callbacks: List[Callback]) -> List[Callback]:
         """
         checkpoints = [c for c in callbacks if isinstance(c, ModelCheckpoint)]
         not_checkpoints = [c for c in callbacks if not isinstance(c, ModelCheckpoint)]
-        return not_checkpoints + checkpoints
+        callbacks = not_checkpoints + checkpoints
```

Comment:
nit: there seems to be some repeated pattern here that could be refactored into its own function